internal/manager/conditions_crd_test.go

```diff
@@ -0,0 +1,75 @@
+package manager_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+)
+
+func TestCRDControllerCondition(t *testing.T) {
+	knownGvr := schema.GroupVersionResource{
+		Group:    "group",
+		Version:  "v1",
+		Resource: "resources",
+	}
+	unknownGVR := schema.GroupVersionResource{
+		Group:    "otherGroup",
+		Version:  "v1",
+		Resource: "resources",
+	}
+
+	restMapper := meta.NewDefaultRESTMapper(nil)
+	restMapper.Add(schema.GroupVersionKind{
+		Group:   knownGvr.Group,
+		Version: knownGvr.Version,
+		Kind:    "Resource",
+	}, meta.RESTScopeRoot)
```

Comment:
```suggestion
	require.NoError(t, restMapper.Add(schema.GroupVersionKind{
		Group:   knownGvr.Group,
		Version: knownGvr.Version,
		Kind:    "Resource",
	}, meta.RESTScopeRoot))
```

Reply:
`Add` doesn't return an error.

Reply:
o.0

https://github.com/Kong/kubernetes-ingress-controller/blob/f4584b309593c92827e9c3adea19e9f020765591/internal/store/store.go#L297

Reply:
It's not `CacheStores`, it's `DefaultRestMapper`:
https://github.com/kubernetes/apimachinery/blob/66e26ac34f07ff0c55347c792157b285b21e5a14/pkg/api/meta/restmapper.go#L99